var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -81,14 +79,14 @@ class Mfem(Package):
 
     def check_variants(self, spec):
         print spec
-        if '+mpi' in spec and ('+hypre' not in spec or '+metis' not in spec):
+        if '+mpi' in spec and ('hypre' not in spec or 'metis' not in spec):
```

Comment:
FYI: you can check this either way.  `+hypre in spec` checks whether the variant is `True`, while `hypre in spec` checks whether `hypre` is a dependency.  They mean the same thing in this build.